psl/psl-core/src/datamodel_connector.rs

```diff
@@ -71,7 +71,17 @@ pub trait Connector: Send + Sync {
     }
 
     /// The referential actions supported by the connector.
-    fn referential_actions(&self, relation_mode: &RelationMode) -> BitFlags<ReferentialAction>;
+    fn referential_actions(&self) -> BitFlags<ReferentialAction>;
+
+    /// The referential actions supported when using relationMode = "prisma" by the connector.
+    /// There are in fact scenarios in which the set of emulated referential actions supported may change
+    /// depending on the connector. For example, Postgres' NoAction mode behaves similarly to Restrict
+    /// (raising an error if any referencing rows still exist when the constraint is checked), but with
+    /// a subtle twist we decided not to emulate: NO ACTION allows the check to be deferred until later
+    /// in the transaction, whereas RESTRICT does not.
```

Comment:
Yep that there are nuances is clear, but we should explain why we want to forbid the emulated referential action in this case (I assume so as not to mislead users into thinking they can get the same behaviour — which they in no case get anyway — but I wasn't part of the discussion)

Reply:
Hopefully my comment response above can clarify this, and we can maybe integrate it into these comments to avoid other misunderstandings 😌